src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -719,11 +703,13 @@ public override Vertex WriteVertex(NodeFactory factory)
 
             VertexBag layoutInfo = new VertexBag();
 
-            KeyValuePair<int, NativeLayoutVertexNode>[] dictionaryEntries = new KeyValuePair<int, NativeLayoutVertexNode>[_genericDictionaryEntries.Count];
+            DictionaryLayoutNode associatedLayout = factory.GenericDictionaryLayout(_method);
+            KeyValuePair<int, NativeLayoutVertexNode>[] dictionaryEntries = new KeyValuePair<int, NativeLayoutVertexNode>[associatedLayout.Entries.Count];
             int index = 0;
-            foreach (KeyValuePair<int, NativeLayoutVertexNode> entry in _genericDictionaryEntries)
+            foreach (GenericLookupResult entry in associatedLayout.Entries)
             {
-                dictionaryEntries[index++] = entry;
+                dictionaryEntries[index] = new KeyValuePair<int, NativeLayoutVertexNode>(index, entry.TemplateDictionaryNode(factory));
+                index++;
             }
             Array.Sort(dictionaryEntries, CompareDictionaryEntries);
```

Comment:
We were getting a list sorted by dictionary entry, but I believe that the dictionary layout should maintain that internally, so I believe we shouldn't need the sort.